modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py

```diff
@@ -333,9 +333,15 @@ def test_read_csv_datetime(
         parse_dates_unsupported = isinstance(parse_dates, dict) or (
             isinstance(parse_dates, list) and isinstance(parse_dates[0], list)
         )
+        if parse_dates_unsupported and engine == "arrow":
+            pytest.skip(
+                "In this specific cases Modin raises `ArrowEngineException` while pandas "
+                "raises `ValueError` that causes tests fails"
+            )
```

Comment:
I think we want to be sure that both of the implementations (modin and pandas) raise an exception in this case, if the sources of these errors are different (because of the different engines: arrow and pandas), we may omit exceptions type checking by passing `check_exception_type=False` to the [`eval_io`](https://github.com/modin-project/modin/blob/421f68c11a6cc7734fa956362250de0002ff3d2e/modin/pandas/test/utils.py#L712)

Reply:
Ok, i see. Case with different exception types was handled more carefully, but some tests still should be skipped because with the given parameters combination Modin implementation raises exception while pandas works fine (that causes tests fail).

Reply:
Do we want to have exceptions for OmniSci backend similar to pandas ones?

Reply:
Not sure that understood your question. In general it is desired option, but the main goal is to keep the same exception type.  Exactly these tests are skipped because Modin [forces](https://github.com/modin-project/modin/blob/master/modin/experimental/engines/omnisci_on_ray/io.py#L310-L311) to raise `ArrowEngineException` exception (without fallback to pandas implementation), that leads to discrepancy with pandas case.

Reply:
> In general it is desired option, but the main goal is to keep the same exception type.

Don't we want to intercept the exception and throw another exception that is concordant with pandas?

Reply:
In the case if we want to achieve the concordance with pandas we just setting `engine = None` (this case is also handled in tests) and after catching of `ArrowEngineException` we are falling back to pandas implementation and getting concordant exception. `arrow` engine is just specific parameter that allows us to check exactly OmniSci function implementation without implicit fallback to pandas.

Reply:
Ok, I see.